python/pyspark/sql/session.py

```diff
@@ -107,10 +126,23 @@ class Builder(object):
         """
 
         _lock = RLock()
-        _options = {}
+        _options = {}  # type: Dict[str, Any]
```

Comment:
Wouldn't be better to use PEP 526 annotations here? 

```python
_options: Dict[str, Any] = {}
```

I doesn't seem we're going to backport hints directly any more, and we're already in Python 3.6 and beyond here.